In [1]:
import pandas as pd
import os
from dataclasses import dataclass, field
import sys
sys.path.insert(0, '/home/jovyan/work')
sys.path.insert(1, '/home/jovyan/work/src')
sys.path.insert(2, '/home/jovyan/work/src/service')

from service import bfs_cases_db_service as bfs_db

from py.global_configs import *
from py.normalize import normalize

/home/jovyan/work/src/models/BfsCase.py:5: SAWarning: On class 'BfsCase', Column object 'drg_cost_weight' named directly multiple times, only one will be used: drg_cost_weight, effective_cost_weight. Consider using orm.synonym instead
  class BfsCase(Base):


In [2]:
# check all the file name

FILES_TO_ANALYZE.keys()


dict_keys(['Hirslanden Salem 2017', 'Hirslanden Beau Site 2017', 'Hirslanden Linde 2017', 'Hirslanden Linde 2018', 'Hirslanden Salem 2018', 'Hirslanden Beau Site 2018', 'Winterthur 2017', 'Winterthur 2018', 'Winterthur 2019', 'Winterthur 2020', 'USZ 2018', 'USZ 2019', 'USZ 2019_30_04_2020', 'KSSG 2021'])

In [3]:
file = FILES_TO_ANALYZE['USZ 2019']
file

FileInfo(path='/home/jovyan/work/src/revised_case_normalization/raw_data/USZ_2018-2019_20200730.xlsx', hospital_name_db='USZ', year='2019', sheets=['Gesamtauffällige_USZ_2019'])

In [4]:

df_revised_case_d2d = normalize(file, 0)

Read 150 cases for USZ 2019
TYPES:
case_id             string
patient_id          string
gender              string
age_years            int64
duration_of_stay     int64
pccl                 int64
drg                 string
pd                  string
bfs_code            string
added_icds          string
removed_icds        string
added_chops         string
removed_chops       string
dtype: object


In [5]:
df_revised_case_d2d.head()

case_id        patient_id gender  age_years  duration_of_stay  pccl  \
0  6400282213  3ED8F10117FCF5F5      M         54                13     3   
1  6400287076  55C5A17F71DADEA0      M         62                 5     3   
2  6400306817  0150AC67E3196B63      W         74                36     3   
3  6400316927  153929A139272585      W         79                 6     3   
4  6400278895  014A89BB816D9215      W         71                44     3   

    drg     pd bfs_code       added_icds removed_icds  \
0  E06A  J4400     M200             F058         <NA>   
1  D62C   J068     M800  I8028,C770,C787         <NA>   
2  G72C    R11     M100              R64         <NA>   
3  B68C  G3531     M100             R471         <NA>   
4  F75D   T827     M200             D508         D648   

                                         added_chops  \
0  99b813::20181227,009a13:L:20181231,948x41::201...   
1                                   890a11::20190211   
2                  948x41::20190607,99c121::20190607   
3                  99c11d::20190708,948x41::20190709   
4                  890a19::20190110,890a32::20181220   

                                       removed_chops  
0  99B813::20181227,009A13:L:20181231,948X41::201...  
1                                   890A11::20190211  
2                  948X41::20190607,99C121::20190607  
3                  99C11D::20190708,948X41::20190709  
4                  890A19::20190110,890A32::20181220

# match to the database


In [6]:
# get the case_id from revised_case

revised_case_id = df_revised_case_d2d['case_id'].values
revised_case_id

array(['6400282213', '6400287076', '6400306817', '6400316927',
       '6400278895', '6400284826', '6400285318', '6400295190',
       '6400285972', '6400296127', '6400287392', '6400305514',
       '6400315570', '6400325817', '6400325819', '6400326160',
       '6400326265', '6400326306', '6400329277', '6400331178',
       '6400331492', '6400331625', '6400332257', '6400334309',
       '6400334358', '6400334421', '6400335091', '6400338217',
       '6400338301', '6400339053', '6400339098', '6400339110',
       '6400339524', '6400343882', '6400330582', '6400330942',
       '6400330984', '6400331207', '6400335350', '6400335613',
       '6400339577', '6400343386', '6400343864', '6400330417',
       '6400330825', '6400331098', '6400342944', '6400327431',
       '6400329606', '6400330783', '6400335620', '6400335765',
       '6400339824', '6400339830', '6400339976', '6400340009',
       '6400343280', '6400343475', '6400325852', '6400325863',
       '6400330299', '6400330346', '6400335461', '64002

In [7]:
# match to the database
revised_case_db = bfs_db.get_bfs_cases_by_ids(revised_case_id)
revised_case_db.head()

drg_cost_weight  aimedic_id  hospital_id     case_id        patient_id  \
0            0.720      337849           14  6400278895  014A89BB816D9215   
1            0.720      337850           14  6400278895  7BCC689D46B39487   
2            1.759      339676           14  6400281744  9E37EB46636D9B84   
3            1.759      339677           14  6400281744  AAAF218127945BA0   
4            0.519      339976           14  6400282164  47AC5130DDC9249B   

   age_years  age_days gender  duration_of_stay  clinic_id  ventilation_hours  \
0         71         0      W                44          4                  0   
1         71         0      W                44          4                  0   
2         86         0      W                12          3                  0   
3         86         0      W                12          3                  0   
4         76         0      M                 8          4                  0   

   admission_weight  gestation_age admission_date admission_type  \
0                 0              0     2018-11-30             11   
1                 0              0     2018-11-30             11   
2                 0              0     2018-12-27             01   
3                 0              0     2018-12-27             01   
4                 0              0     2018-12-26             06   

  discharge_date discharge_type   drg adrg  pccl  
0     2019-01-13             00  F75D  F75     3  
1     2019-01-13             00  F75D  F75     3  
2     2019-01-08             00  F59C  F59     3  
3     2019-01-08             00  F59C  F59     3  
4     2019-01-03             00  X60C  X60     2

In [8]:
# 
print('There are {} out of {} revised cases from DtoD that are matched with the database for {} {}'.format(len(revised_case_db), len(revised_case_id), file.hospital_name_db, file.year))

There are 224 out of 150 revised cases from DtoD that are matched with the database for USZ 2019


### Attention, there are duplicated case_id in the USZ dataset
e.g. 6400278895 has two patient_id

In [9]:
# if we find the match cases, then we need to check if the case_id, gender, year....are match